integration/simulation/simulation_test.go

```diff
@@ -30,21 +34,62 @@ func TestSimulation(t *testing.T) {
 	defer f.Close()
 	common.SetLog(f)
 
-	blockDuration := uint64(20_000)
-	l1netw, l2netw := RunSimulation(5, 10, 15, blockDuration, blockDuration/15, blockDuration/3)
-	firstNode := l2netw.nodes[0]
-	checkBlockchainValidity(t, l1netw, l2netw, firstNode.Enclave.TestDB(), firstNode.Enclave.TestPeekHead().Head)
-	stats := l1netw.Stats
+	txGenerator := NewTransactionGenerator(5)
+
+	numberOfNodes := 10
+	simulationTime := 15
+	avgBlockDuration := uint64(20_000)
+	avgLatency := avgBlockDuration / 15
+	avgGossipPeriod := avgBlockDuration / 3
+
+	stats := NewStats(numberOfNodes, simulationTime, avgBlockDuration, avgLatency, avgGossipPeriod)
+
+	l1Network, l2Network := RunSimulation(txGenerator, numberOfNodes, simulationTime, avgBlockDuration, avgBlockDuration/15, avgBlockDuration/3, stats)
+
+	checkBlockchainValidity(t, l1Network, l2Network, txGenerator)
```

Comment:
no need to pass the networks, since they're part of the generator anyway

Reply:
Refactored this further, worth having another look :)